packages/flutter/lib/src/painting/borders.dart

```diff
@@ -259,6 +259,318 @@ class BorderSide {
   String toString() => '$runtimeType($color, ${width.toStringAsFixed(1)}, $style)';
 }
 
+/// Base class for shape outlines.
+///
+/// This class handles how to add multiple borders together.
+@immutable
+abstract class ShapeBorder {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const ShapeBorder();
+
+  /// The widths of the sides of this border represented as an [EdgeInsets].
+  ///
+  /// Specifically, this is the amount by which a rectangle should be inset so
+  /// as to avoid painting over any important part of the border. It is the
+  /// amount by which additional borders will be inset before they are drawn.
+  ///
+  /// This can be used, for example, with a [Padding] widget to inset a box by
+  /// the size of these borders.
+  ///
+  /// Shapes that have a fixed ratio regardless of the area on which they are
+  /// painted, or that change their rendering based on the size, for instance
+  /// [CircleBorder], will not return invalid [dimensions] information (because
+  /// they cannot know their eventual size when computing their dimensions).
```

Comment:
it's not really "their" size. ShapeBorders don't have a size, they're just told to paint at a size.

Reply:
Went with:
```
  /// Shapes that have a fixed ratio regardless of the area on which they are
  /// painted, or that change their rendering based on the size they are given
  /// when painting (for instance [CircleBorder]), will not return valid
  /// [dimensions] information because they cannot know their eventual size when
  /// computing their [dimensions].
```